addon/components/flexberry-file.js

```diff
@@ -1,334 +1,523 @@
-   * Flag: indicates whether to show modal dialog on upload errors or not.
+    Flag: indicates whether to show modal dialog on upload errors or not.
+
+    @property showModalDialogOnUploadError
+    @type Boolean
+    @default false
    */
   showModalDialogOnUploadError: undefined,
 
   /**
-   * Flag: indicates whether to show modal dialog on download errors or not.
-   */
-  showModalDialogOnDownloadError: undefined,
+    Caption to be displayed in error modal dialog.
+    It will be displayed only if some error occur.
 
-  /**
-   * Title to be displayed in error modal dialog.
-   * It will be displayed only if some error occur.
-   *
-   * @property errorModalDialogTitle
-   * @type String
-   * @default 't('components.flexberry-file.error-dialog-title')'
+    @property errorModalDialogCaption
+    @type String
+    @default 't('components.flexberry-file.error-dialog-caption')'
    */
-  errorModalDialogTitle: t('components.flexberry-file.error-dialog-title'),
+  errorModalDialogCaption: t('components.flexberry-file.error-dialog-caption'),
 
   /**
-   * Content to be displayed in error modal dialog.
-   * It will be displayed only if some error occur.
-   *
-   * @property errorModalDialogContent
-   * @type String
-   * @default 't('components.flexberry-file.error-dialog-content')'
+    Content to be displayed in error modal dialog.
+    It will be displayed only if some error occur.
+
+    @property errorModalDialogContent
+    @type String
+    @default 't('components.flexberry-file.error-dialog-content')'
    */
   errorModalDialogContent: t('components.flexberry-file.error-dialog-content'),
 
   /**
-   * Selected jQuery object, containing HTML of error modal dialog.
+    Selected jQuery object, containing HTML of error modal dialog.
+
+    @property errorModalDialog
+    @type Object
+    @default null
    */
   errorModalDialog: null,
 
-  /**
-   * File name.
-   * It is binded to component file name input, so every change to fileName will automatically change file name input value.
-   */
-  fileName: Ember.computed('value', function() {
-    var jsonValue = this.get('jsonValue');
-    if (Ember.isNone(jsonValue)) {
-      return null;
-    }
+  actions: {
+    /**
+      Handles click on selected image preview and sends action with data outside component
+      in order to view selected image at modal window.
 
-    return jsonValue.fileName;
-  }),
+      @method actions.viewLoadedImage
+      @public
+     */
+    viewLoadedImage() {
+      let fileName = this.get('fileName');
+      let previewImageAsBase64String = this.get('_previewImageAsBase64String');
+      if (!Ember.isBlank(fileName) && !Ember.isBlank(previewImageAsBase64String)) {
+        this.sendAction('viewImageAction', {
+          fileSrc: previewImageAsBase64String,
+          fileName: fileName
+        });
+      }
+    },
+
+    /**
+      Handles click on add button.
+
+      @method actions.addButtonClick
+      @public
+     */
+    addButtonClick() {
```

Comment:
For now it should be empty, because button is attached to file input through label's 'for' attrubute,
but i don't' think that it is necessary to remove this handler, code is cleaner with it.
I'll add comment into addButtonClick method.